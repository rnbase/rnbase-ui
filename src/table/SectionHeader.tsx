import React from 'react';
import { StyleProp, StyleSheet, Text, TextStyle, View, ViewProps } from 'react-native';

import { Theme, Themed, withTheme } from '../theming';

interface SectionHeaderProps extends ViewProps {
  children?: React.ReactNode;
  text?: string;
  textStyle?: StyleProp<TextStyle>;
}

const SectionHeader: React.FC<Themed<typeof createStyleSheet, SectionHeaderProps>> = ({
  theme: { styles },
  children,
  text,
  style,
  textStyle,
  ...props
}) => (
  <View {...props} style={[styles.root, style]}>
    {children || (
      <Text style={[styles.text, textStyle]} numberOfLines={1}>
        {text}
      </Text>
    )}
  </View>
);

const createStyleSheet = ({ colors, fonts }: Theme) =>
  StyleSheet.create({
    root: {
      marginBottom: 6.5,
      marginHorizontal: 20,
    },
    text: {
      ...fonts.normal,
      fontSize: 13,
      color: colors.gray,
      textTransform: 'uppercase',
    },
  });

export default withTheme(SectionHeader, createStyleSheet, 'Table.SectionHeader');
