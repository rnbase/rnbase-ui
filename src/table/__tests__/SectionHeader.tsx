import React from 'react';

// Note: test renderer must be required after react-native.
import TestRenderer from 'react-test-renderer';

/**
 * Under test
 */
import SectionHeader from '../SectionHeader';

type Props = React.ComponentProps<typeof SectionHeader>;

const Component = (props: any) => React.createElement('Component', props);

const createElement = (props: Partial<Props>) => <SectionHeader {...props} />;

const createRenderer = (props: Partial<Props>) => TestRenderer.create(createElement(props));

it('should render normally as text', () => {
  expect(createRenderer({ children: 'TEXT' })).toMatchSnapshot();
});

it('should render normally as component', () => {
  expect(createRenderer({ children: <Component /> })).toMatchSnapshot();
});

it('should render normally as text with custom styles', () => {
  expect(createRenderer({ children: 'TEXT', style: { color: 'red' } })).toMatchSnapshot();
});

it('should render normally as component with custom styles', () => {
  expect(createRenderer({ children: <Component />, style: { color: 'red' } })).toMatchSnapshot();
});
