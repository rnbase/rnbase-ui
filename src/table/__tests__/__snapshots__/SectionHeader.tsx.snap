// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render normally as component 1`] = `
<View
  style={
    Array [
      Object {
        "marginBottom": 6.5,
        "marginHorizontal": 20,
      },
      undefined,
    ]
  }
>
  <Component />
</View>
`;

exports[`should render normally as component with custom styles 1`] = `
<View
  style={
    Array [
      Object {
        "marginBottom": 6.5,
        "marginHorizontal": 20,
      },
      Object {
        "color": "red",
      },
    ]
  }
>
  <Component />
</View>
`;

exports[`should render normally as text 1`] = `
<View
  style={
    Object {
      "marginBottom": 6.5,
      "marginHorizontal": 20,
    }
  }
>
  <Text
    numberOfLines={1}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "color": "#838387",
          "fontFamily": "System",
          "fontSize": 13,
          "fontWeight": "400",
          "includeFontPadding": false,
          "textAlignVertical": "center",
          "textTransform": "uppercase",
        },
        undefined,
      ]
    }
  >
    TEXT
  </Text>
</View>
`;

exports[`should render normally as text with custom styles 1`] = `
<View
  style={
    Object {
      "marginBottom": 6.5,
      "marginHorizontal": 20,
    }
  }
>
  <Text
    numberOfLines={1}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "color": "#838387",
          "fontFamily": "System",
          "fontSize": 13,
          "fontWeight": "400",
          "includeFontPadding": false,
          "textAlignVertical": "center",
          "textTransform": "uppercase",
        },
        Object {
          "color": "red",
        },
      ]
    }
  >
    TEXT
  </Text>
</View>
`;
